ql/src/java/org/apache/hadoop/hive/ql/exec/OrcFileMergeOperator.java

```diff
@@ -113,7 +116,11 @@ private void processKeyValuePairs(Object key, Object value)
 
       // store the orc configuration from the first file. All other files should
       // match this configuration before merging else will not be merged
-      if (outWriter == null) {
+      int bucketId = 0;
+      if (conf.getIsCompactionTable()) {
```

Comment:
Can we do this outside the process loop? Maybe saving some time, since this part of the code runs for every item, and this can cause performance degradation (I guess)

Reply:
Ok.. as discussed this was an overreaction 😄